### PR TITLE
Kam shtuat funksionin per dekodimin e jwt tokenit per marrjen e userR…

### PR DESCRIPTION
**Pull request by RedonAdemi**
…olit pas marrjes se rolit nga tokeni e riderecton user ne dashboardin e userolit qe i takon -Kam thjeshtesuar dhe organizuar kodin -Kam shtuar function makeAuthenticatedRequest i cili eshte funksioni i riperdoreshem Accepts URL and Method,Checks Authentication,Makes AJAX Request,Handles Success and Error

### CODE REVIEW

File: ReviveIT/WebUI/Controllers/CompanyController.cs [hunk 1]
```diff
@@ -1,23 +1,19 @@
...
+using System.Security.Claims;
 
-//[Authorize(Roles = "Company")] - for authorize and ViewBag for testing
-
-[AllowAnonymous]
-[Route("Company")]
 public class CompanyController : Controller
 {
-    [HttpGet("Index")]
+
```
Comment (donatgosalcii) [inline ReviveIT/WebUI/Controllers/CompanyController.cs:8]: remove white line

File: ReviveIT/WebUI/Controllers/CompanyController.cs [hunk 2]
```diff
@@ -1,23 +1,19 @@
...
+ 
     public IActionResult Index()
     {
         ViewBag.Role = "Company";
         return View();
     }
 
-    [HttpGet("Inbox")]
+    [Authorize(Roles = "Company")]
+
```
Comment (donatgosalcii) [inline ReviveIT/WebUI/Controllers/CompanyController.cs:16]: No need for a while line

File: ReviveIT/WebUI/Controllers/CustomerController.cs [hunk 3]
```diff
@@ -1,44 +1,39 @@
...
-using Microsoft.AspNetCore.Mvc;
+using System.Security.Claims;
 
-//[Authorize(Roles = "Customer)] - for authorize
-[AllowAnonymous]
-[Route("Customer")]
 public class CustomerController : Controller
 {
-    [HttpGet("Index")]
+
```
Comment (donatgosalcii) [inline ReviveIT/WebUI/Controllers/CustomerController.cs:8]: remove white line

File: ReviveIT/WebUI/Controllers/CustomerController.cs [hunk 4]
```diff
@@ -1,44 +1,39 @@
...
-    [HttpGet("TechniciansCompanies")]
+    [Authorize(Roles = "Customer")]
     public IActionResult TechniciansCompanies()
     {
         ViewBag.Role = "Customer";
         return View();
     }
 
-    [HttpGet("PostJob")]
+
```
Comment (donatgosalcii) [inline ReviveIT/WebUI/Controllers/CustomerController.cs:22]: remove w line

File: ReviveIT/WebUI/Controllers/TechnicianController.cs [hunk 5]
```diff
@@ -1,34 +1,32 @@
...
-    //[Authorize(Roles = "Company")] - for authorize and ViewBag for testing
-    [AllowAnonymous]
-    [Route("Technician")]
-    public class TechnicianController : Controller
-    {
-        [HttpGet("Index")]
-        public IActionResult Index()
+public class TechnicianController : Controller
+{
+
```
Comment (donatgosalcii) [inline ReviveIT/WebUI/Controllers/TechnicianController.cs:8]: w line

File: ReviveIT/WebUI/program.cs [hunk 6]
```diff
@@ -2,16 +2,20 @@
...
 using Infrastructure.Services;
+using Microsoft.AspNetCore.Authentication.Cookies;
 using Microsoft.AspNetCore.Authentication.JwtBearer;
 using Microsoft.AspNetCore.Identity;
 using Microsoft.EntityFrameworkCore;
 using Microsoft.IdentityModel.Tokens;
 using System.Text;
 using WebUI.MiddleWares;
 
+
```
Comment (donatgosalcii) [inline ReviveIT/WebUI/program.cs:17]: why 2 white lines here

**Review by NathanAmanuelG — APPROVED** [review]
from what I can see, your changes are all positive and should work as intended. great work!

File: ReviveIT/WebUI/program.cs [hunk 7]
```diff
@@ -37,6 +41,8 @@
     options.AddPolicy("RequireAuthenticatedUser", policy => policy.RequireAuthenticatedUser());
 });
 
+
+
```
Comment (donatgosalcii) [inline ReviveIT/WebUI/program.cs:45]: remove this and 44

File: ReviveIT/WebUI/program.cs [hunk 8]
```diff
@@ -75,9 +81,14 @@
 builder.Services.AddHttpContextAccessor();
 
 builder.Services.AddSingleton<ConfigurationConstant>();
+builder.Services.AddAuthorization(options => { options.AddPolicy("TechnicianOnly", policy => policy.RequireRole("Technician")); });
+builder.Services.AddAuthorization(options => { options.AddPolicy("CustomerOnly", policy => policy.RequireRole("Customer")); });
+builder.Services.AddAuthorization(options => { options.AddPolicy("CompanyOnly", policy => policy.RequireRole("Company")); });
+
```
Comment (donatgosalcii) [inline ReviveIT/WebUI/program.cs:87]: no need for white line

File: ReviveIT/WebUI/program.cs [hunk 9]
```diff
@@ -75,9 +81,14 @@
...
 
 var app = builder.Build();
 
+
```
Comment (donatgosalcii) [inline ReviveIT/WebUI/program.cs:91]: this too

File: ReviveIT/WebUI/program.cs [hunk 10]
```diff
@@ -106,6 +117,7 @@
 app.UseAuthorization();
 app.UseMiddleware<ExceptionMiddleware>();
 
+
```
Comment (donatgosalcii) [inline ReviveIT/WebUI/program.cs:120]: remove useless white lines

File: ReviveIT/WebUI/wwwroot/js/login.js [hunk 11]
```diff
@@ -1,4 +1,10 @@
 ﻿$(document).ready(function () {
+    // Function to check if the user is on the login page
```
Comment (donatgosalcii) [inline ReviveIT/WebUI/wwwroot/js/login.js:2]: remove comments from this file too

File: ReviveIT/WebUI/wwwroot/js/login.js [hunk 12]
```diff
@@ -1,4 +1,10 @@
...
+    function isLoginPage() {
+        return window.location.pathname === "/logIn/LogIn";
+    }
+
```
Comment (donatgosalcii) [inline ReviveIT/WebUI/wwwroot/js/login.js:6]: white lines aswell

**Review by donatgosalcii — CHANGES_REQUESTED** [review]
Review your changes for white lines and comments, code should be clean of them!

**Review by NathanAmanuelG — COMMENTED** [review]
Sorry donat but I just saw that this pr would delete the login.js whilst the other one would conflict with, why not do one pr with both changes.

**Pull request closed without merging**